aptos-indexer-processors-sdk/testing-framework/src/mock_grpc.rs

```diff
@@ -27,31 +27,45 @@ impl RawData for MockGrpcServer {
         &self,
         req: Request<GetTransactionsRequest>,
     ) -> Result<Response<Self::GetTransactionsStream>, Status> {
-        let version = req.into_inner().starting_version.unwrap();
-
-        // Find the specific transaction that matches the version
-        let transaction = self
-            .transactions_response
-            .iter()
-            .flat_map(|transactions_response| transactions_response.transactions.iter())
-            .find(|tx| {
-                tx.version == version // Return the transaction that matches the version
-            });
-
-        let result = match transaction {
-            Some(tx) => {
-                // Build a new TransactionResponse with this matching transaction
-                TransactionsResponse {
-                    transactions: vec![tx.clone()],
-                    chain_id: Some(self.chain_id),
-                }
-            },
-            None => {
-                // No matching transaction found, return a default response with the first transaction
-                let mut default_transaction_response = self.transactions_response[0].clone();
-                default_transaction_response.chain_id = Some(self.chain_id); // Set the chain_id field
-                default_transaction_response
-            },
+        let request = req.into_inner();
+        let starting_version = request.starting_version.unwrap();
+        let transactions_count = request.transactions_count.unwrap_or(1); // Default to 1 if transactions_count is not provided
+
+        // Collect transactions starting from `starting_version`, without any gaps, up to `transactions_count`.
+        let mut collected_transactions = Vec::new();
+        let mut current_version = starting_version;
+
+        // Step 1: Build a map of transactions keyed by version for quick access
+        let mut transaction_map = HashMap::new();
+        for transaction_response in &self.transactions_response {
+            for tx in &transaction_response.transactions {
+                transaction_map.insert(tx.version, tx);
+            }
+        }
+
+        // Step 2: Collect transactions in the correct sequence starting from `starting_version`
+        while collected_transactions.len() < transactions_count as usize {
+            if let Some(tx) = transaction_map.get(&current_version) {
+                collected_transactions.push((*tx).clone()); // Collect the transaction
+                current_version += 1; // Move to the next expected version
+            } else {
+                // If no transaction is found for the current version, stop looking for more
+                break;
+            }
+        }
```

Comment:
Will this work if the test transactions in json have gaps? I guess we could return them sorted by version?

Reply:
wouldn't it break if we have gaps and return only what we have? What's the behavior that sdk processor  expects from the grpc?

and return them sored by version will make the validation logic simpler though

Reply:
So `TransactionStreamInternal` has logic that asserts there's no gaps in between `TransactionResponse` batches: [link](https://github.com/aptos-labs/aptos-indexer-processors-sdk/blob/343a2d01dce9f8bca76b90d506f61c1a07943e4b/aptos-indexer-processors-sdk/transaction-stream/src/transaction_stream.rs#L429)

Within a batch, there's no gap checking, so we could return all the transactions sorted inside of 1 `TransactionResponse` and that would get around the gap check.

Reply:
Between SDK steps, there's no guarantee that the input and output batches are sorted by version. So that's why we needed the OrderByVersion step if you wrote one of the step in a way that you assume the batches are sorted.

Reply:
> Will this work if the test transactions in json have gaps?
when setting up test_context with inputs with gaps, processor would request for 1 txn at a time, otherwise it won't be able to get transactions.

Reply:
Hmm how come it wouldn't work? What if you put more than 1 transaction in the vec inside of TransactionResponse?

Reply:
oh now i understand your question. you could add any number of txns in the vec inside of TransactionResponse with gaps 
but the processor config should be constructed in a way that it only requests with the matching starting_version and txn_count. so that it doesn't try to poll the data with gaps.

Reply:
Yeah I think that could work! The starting_version and txn_count can be derived from self.transactions_response

Reply:
@yuunlimm would you mind adding a comment pointing out this nuance about multiple transactions/how the config should be constructed? this was confusing to me

Reply:
Should we have a helper func to construct the txn stream config for them, to avoid any bugs or confusion? The config is very much coupled with how the grpc stream is mocked.